### PR TITLE
Add initial install diagrams for style and content suggestions

### PR DESCRIPTION
**Pull request by ousbots**
#### What this PR does / why we need it:

This PR is for discussing the kots install diagrams.

#### Which issue(s) this PR fixes:

No issue, this is to hopefully help people understand kots workflows better.

#### Special notes for your reviewer:

These are incomplete and I'm putting them up here as an example for critique before spending too much more time finishing the diagrams. It took me a while to figure out a good way to display an appropriate level of detail and context without being mostly noise.

There are two charts here. The first is a sequence diagram of the communication between kots, k8s, docker, and kotsadm parts. The second is a higher level flow chart of the kots cli install process.

## Steps to reproduce

To rebuild the images:
```
cd kots/design/diagrams
yarn install
yarn run build
```

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Does this PR require documentation?

```
NONE
```

### CODE REVIEW

**Comment by ousbots** [comment]
What do you think about the level of detail in these diagrams? Would it be useful to indicate better which function and file each step is in, maybe with boxed and colored backgrounds with this info? How about the presentation of what data kots sends to k8s (and other components) in the sequence diagram, does that make sense? Would more notes describing higher level process and the reason why it's happening help or just be clutter?

**Comment by ousbots** [comment]
Also, these are very incomplete as they're the first iteration of the diagrams that I thought weren't confusing and way too busy

**Comment by ousbots** [comment]
What about the level of Kubernetes details? I wasn't sure if I should explain what each Kubernetes request is doing and why it's used in the context of an install. Is everyone familiar enough with what it means to create things like configmaps, secrets, namespaces, deployments, and services?

File: design/diagrams/kots_install_sequence.mmd [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+	participant k8s as Cluster k8s API
+	participant kotsadm as Kots Admin API
+	participant kotsdb as Kots Admin Database (rqlite)
+	participant kotsstore as Kots Admin Data Store (minio)
+	participant docker as Docker Registry
+	participant repli as Replicated API
+
+	%% Check that there are sufficient RBAC permissions on the cluster
+	note over kots, k8s: Check that there are sufficient priveledges on the cluster to deploy the app
+	kots -> k8s: Self Subject Access Review of resource attributes [Verb: *, Group: *, Version: *, Resource: *]
```
Comment (rianfowler) [inline design/diagrams/kots_install_sequence.mmd:12]: `->` doesn't render an arrowhead; `->>` and `-->>` [will](https://mermaid.js.org/syntax/sequenceDiagram.html#messages) though.

**Review by rianfowler — APPROVED** [review]
this is pretty good. thanks for putting it together.

**Pull request closed without merging**